x-pack/legacy/plugins/infra/public/pages/logs/analysis/page.tsx

```diff
@@ -30,6 +31,12 @@ export const AnalysisPage = () => {
     spaceId,
     timeField: source ? source.configuration.fields.timestamp : '',
   });
+  const { cleanupMLResources, isCleaningUp } = useLogAnalysisCleanup({ sourceId, spaceId });
+  useEffect(() => {
+    if (didSetupFail) {
+      cleanupMLResources();
```

Comment:
looks like `didSetupFail` returns true even if the `setupMlModuleRequest.state == "resolved"` because it's only checking if it's `"uninitialized"` in `log_analysis_jobs.tsx`

i.e. it deletes the job every time

Reply:
This isn’t ready for review yet, I ran in to further ML errors. But, the original issue here was that there actually seems to be a timing issue in `useTrackedPromise`, one of the memoized values was reacting to the request state change, causing `didSetupFail` to switch into a temporary bad state, causing the delete, with the call to `onResolve` following up later. Anyways, I’ve changed the way `didSetupFail` Works to use direct failures instead as it’s simpler. Although I need to tweak that to account for retrying more than once, as right now the dependencies will just see the same value and not re-run.

Reply:
> This isn’t ready for review yet

Aw sorry, draft PR. Just wanted to give feedback in case it was helpful but feel free to ignore if you're still in flux. 🙏

Reply:
Np, just didn't want you wasting review beans.